003_data_logging/003_data_logging.tex

```diff
@@ -0,0 +1,86 @@
+\chapter{Data Logging and Dashboard}\label{data_logging}
+
+\section{Why do we need data?}
+
+Data logging enables analysis of how internal variables change with time during a simulation. One might be interested in different variables from the system in question: messages frequency, food variation over time, agent deaths rate, etc... Analysis also requires linking different simulation parameters with the data logged (e.g. the number of agents or the amount of food available). 
+
+It is therefore important to setup a logging framework that structures data effectively and scales efficiently for large simulations. The below subsections will introduce the simulation parametrisation, followed by the different logging strategies (dumps and structured), and concluding with the logging interface and the analysis dashboard.
+
+\section{Parametrisation}
+
+First, it was important to enable an effective parametrisation of the simulation. A configuration interface was therefore set to abstract variables from the simulation environment. A configuration object would therefore be passed to the simulation environment instance, and it would define the following simulation parameters:
```

Comment:
```suggestion
First, it was important to enable an effective parametrisation of the simulation. A configuration interface was therefore set to abstract variables from the simulation environment and a configuration object would subsequently be passed to the simulation environment instance to define the following simulation parameters:
```

Again, let me know if this is better. Prevents the staggered sentences and repetition of 'therefore'